pkg/metrics/process/sampler_darwin.go

```diff
@@ -38,17 +39,21 @@ func NewProcessSampler(ctx agent.AgentContext) sampler.Sampler {
 
 	ttlSecs := config.DefaultContainerCacheMetadataLimit
 	apiVersion := ""
-	interval := config.FREQ_INTERVAL_FLOOR_PROCESS_METRICS
 	dockerContainerdNamespace := ""
+	interval := config.FREQ_INTERVAL_FLOOR_PROCESS_METRICS
+	var containerSamplers []metrics.ContainerSampler
 	if hasConfig {
 		cfg := ctx.Config()
 		ttlSecs = cfg.ContainerMetadataCacheLimit
 		apiVersion = cfg.DockerApiVersion
 		dockerContainerdNamespace = cfg.DockerContainerdNamespace
 		interval = cfg.MetricsProcessSampleRate
+
+		if cfg.ProcessContainerDecoration {
```

Comment:
Is the default value of `ProcessContainerDecoration` (true) be honoured if no config is added?

Reply:
Yes, it has a default value https://github.com/newrelic/infrastructure-agent/pull/1888/files#diff-2658370b0497fa11f0682953637e6e2d237a568851f410b6de8e1ad202045a6cR116

Reply:
oh so this `if` block is kind of useless i guess, `hasConfig` will be always `true` whatever the situation then?

Reply:
There is no real use case or way to call this without config. I guess that this check was added as AgentContext is an Interface and config is a pointer. So this makes it secure from execution pov.

(in any case, I don't want to change the behavior now 😅 )